packages/browser/__tests__/Session.spec.ts

```diff
@@ -559,10 +563,16 @@ describe("Session", () => {
         .fn()
         .mockResolvedValue(undefined);
       const mySession = new Session({ clientAuthentication });
-      await mySession.handleIncomingRedirect({
+      // eslint-disable-next-line no-void
+      void mySession.handleIncomingRedirect({
         url: "https://some.redirect/url",
         restorePreviousSession: true,
       });
+      // We know that handleincomingRedirect is never going to return, so just
+      // wait some time for the async calls to complete.
+      await new Promise((resolve) => {
+        setTimeout(resolve, 500);
+      });
```

Comment:
I don't understand what's going on here... Is the problem that it asynchronously writes to local storage, so we'll have to be able to await that in some way? The "some time" and `useRealTimers` here makes me nervous - wouldn't it just be enough to wait for the next tick in the event loop, i.e. `await Promise.resolve()`?

Reply:
The issue here is that ` void mySession.handleIncomingRedirect(...)` only resolves if no redirect is happening, which means that it does not resolve in "normal" conditions, so that the client code which looks like `await session.login()` is actually blocked until the actual redirect happens.

This piece of code is intended to work around this never returning promise. On one hand, we can't await it, but on the other some events in ` void mySession.handleIncomingRedirect(...)` are actually asynchronous, so not acknowledging that could introduce race conditions in the tests. I didn't realize using `await Promise.resolve()` could be a good fix for this, I'll try it out.

Reply:
Unfortunately, it isn't. My assumption is that `await Promise.resolve()` skips one tick on the event loop, but that  asynchronous operations happen for several ticks (I'm not very familiar with how the NodeJS event loop works, so if what I say doesn't make sense that's why ^^). I tested locally "sleeping" for 50ms instead of 500 ms, and it worked too, but my thinking was that putting a long wait is sure not to rely on the computer running the tests being actually quick, and even some old tired laptop can clear the bar easy ^^

Reply:
Right, so it were multiple Promises that were being awaited, so that's indeed multiple ticks. I've added a commit that explicitly awaits those Promises, as a suggestion: 3d2cd3df74a172046a3f7ff6781920edc1778a0c

Reply:
I see, yeah that looks much better than waiting for an arbitrary duration ! Thanks.